### ISSUE TITLE
Inject view in MVP architecture

### ISSUE THREAD

**Issue by jablonskipj** [Ekito/koin#62]
Hello,

I have a problem with the implementation of MVP pattern using KOIN. I would like to inject `view` into `presenter`.  My view is extending fragment and I do not know how to provide it to Presenter object while injection. I keep receiving info `org.koin.error.BeanInstanceCreationException: Can't create bean Bean[class=com.pleez.fragment.sentlist.SentListContract.Presenter] due to error :
	org.koin.error.NoBeanDefFoundException: No bean definition found for class interface com.pleez.fragment.sentlist.SentListContract$View (Kotlin reflection is not available)`. 

Here is how I have declared my module:

```kotlin
val sentListModule: Module = applicationContext {
    provide { SentListPresenter(get(), get()) as SentListContract.Presenter }
    provide { FragmentSentList() as SentListContract.View }
    provide { SentListDataSource(get(), get()) as SentListContract.DataSource }
}
```
can someone tell me what is missing, or what I have done wrong?

**Comment by arnaudgiuliani** [comment]
Hello,

it's a weird case.  It doesn't find your `SentListContract.View` component. Can you share a bit more? ( FragmentSentList class ...)

**Comment by arnaudgiuliani** [comment]
What version of Koin do you use?

**Comment by jablonskipj** [comment]
So, I have an application which code remind more spaghetti (I have inherited it after developer from India), and I am afraid that `FragmentSentList` is a bit too long to share here. I will describe a little what I have done. 

`FragmentSentList` is java class that extends `SentListView`. `SentListView` class implementation looks like this:

```kotlin
open class SentListView: Fragment(), SentListContract.View {
    val presenter: SentListContract.Presenter by inject()
}
```

and my `FragmentSentList` looks like this:

```java
public class FragmentSentList extends SentListView implements SentListSearchAdapter.onItemClick {
 ... body
}
```
so I have created kotlin code, to enable Koin injection into java code. Maybe this is the case?

**Comment by arnaudgiuliani** [comment]
The problem of MVP is the two way binding between view and model. Here below 2 apps that you may check to find more clues:
* https://github.com/Ekito/android-architecture/tree/dev-todo-mvp-kotlin-koin/todoapp
* https://github.com/Ekito/koin-samples/tree/master/samples/android-weatherapp

This means that you have to take the `FragmentSentList` instance from Koin too. You use something like `val ... : SentListContract.View by inject()` somewhere?

**Comment by jablonskipj** [comment]
no, I do not use `val ... : SentListContract.View by inject()`. I wanted to provide it to presenter by constructor like: 
```kotlin
class SentListPresenter(val view: SentListContract.View ,val dataSource: SentListContract.DataSource): SentListContract.Presenter {
```

**Comment by arnaudgiuliani** [comment]
Come back here if you still need some help

**Comment by jablonskipj** [comment]
I have taken a look at implementations that are done in projects that you passed here. There is a similar workaround as I have prepared. So those examples are not explaining to me how to inject View by constructor but show me that others do this in the same way as I did. Here is a code that shows how the view is provided to the presenter in `todo app`:

```kotlin
   override fun onResume() {
        super.onResume()
        presenter.view = this
        presenter.start()
    }
```

**Comment by arnaudgiuliani** [comment]
Yes, we usually make the MVP injection like this. However, in next release of Koin we will be able to pass a parameter for injection. Check last comments from https://github.com/Ekito/koin/issues/49
